src/main/java/org/agbrothers/englishbot/service/LessonService.java

```diff
@@ -4,38 +4,90 @@
 import org.agbrothers.englishbot.entity.Word;
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.security.SecureRandom;
+import java.util.*;
 
 @Service
 public class LessonService {
 
     private final Map<String, Lesson> lessonRegistry;
     private final WordPoolService wordPoolService;
+    private int countCorrectAnswers = 0;
+    private int countIncorrectAnswer = 0;
+    private Word currentWord;
```

Comment:
This will not work if you have multiple users, since `LessonService` is a Spring singleton so that this `currentWord` reference will be the same for all of the threads. I.e. If you and I will start the lesson simultaneously, we will share the same current word.

Long story short: you should leave `currentWord` reference in `Lesson`, and pass the `lesson` object to `LessonService` methods